File: cpp/08_003.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

std::vector<std::vector<std::string>> groupWordPatterns(std::vector<std::string>& words) {
    std::unordered_map<std::string, std::vector<std::string>> groups;
    
    for (const std::string& word : words) {
        // Generate key by sorting characters
        std::string key = word;
        std::sort(key.begin(), key.end());
        
        // Add word to its group
        groups[key].push_back(word);
    }
    
    // Return lists of grouped words
    std::vector<std::vector<std::string>> result;
    for (const auto& pair : groups) {
        result.push_back(pair.second);
    }
    
    return result;
}

File: data structures and algorithms/cs/08_003.cs
using System;
using System.Collections.Generic;

public List<List<string>> GroupWordPatterns(string[] words) {
    Dictionary<string, List<string>> groups = new Dictionary<string, List<string>>();
    
    foreach (string word in words) {
        // Generate key by sorting characters
        char[] chars = word.ToCharArray();
        Array.Sort(chars);
        string key = new string(chars);
        
        // Add word to its group
        if (!groups.ContainsKey(key)) {
            groups[key] = new List<string>();
        }
        groups[key].Add(word);
    }
    
    // Return lists of grouped words
    return new List<List<string>>(groups.Values);
}